### PR TITLE
feat: convert action builders to classes

### PR DESCRIPTION
**Pull request by robercano**
- Converted all action builders to classes
- Added convenience functions like `_delegateToProtocol` and `_getContractAddress`
- Included information on the actions that they are using for aiding the operation definition tool
- Updated all tests

### CODE REVIEW

File: sdk/protocol-plugins/src/plugins/aave-v3/builders/AaveV3DepositBorrowActionBuilder.ts [hunk 1]
```diff
@@ -3,91 +3,100 @@ import {
...
+  /**
+   * Resolves the target address for the borrow action based on the borrow target type
+   * @param params The parameters for the action builder
+   * @returns The address of the target contract
+   */
+  private async _getBorrowTargetAddress(
+    params: ActionBuilderParams<steps.DepositBorrowStep>,
+  ): Promise<IAddress> {
+    const { user, step, positionsManager, addressBookManager } = params
+    if (step.inputs.borrowTargetType === TokenTransferTargetType.PositionsManager) {
```
Comment (halaprix) [inline sdk/protocol-plugins/src/plugins/aave-v3/builders/AaveV3DepositBorrowActionBuilder.ts:92]: maybe check explicitely for other `TokenTransferTargetType`s and throw if it's not known

File: sdk/protocol-plugins/src/implementation/BaseProtocolPlugin.ts [hunk 2]
```diff
@@ -136,8 +136,14 @@ export abstract class BaseProtocolPlugin implements IProtocolPlugin {
...
-    return this.stepBuilders[step.type] as ActionBuilder<T>
+  getActionBuilder<StepType extends steps.Steps>(step: StepType): Maybe<IActionBuilder<StepType>> {
+    const builder = this.stepBuilders[step.type]
+
+    if (!builder) {
+      return undefined
+    }
+
+    return new builder() as IActionBuilder<StepType>
   }
```
Comment (halaprix) [inline sdk/protocol-plugins/src/implementation/BaseProtocolPlugin.ts:147]: consider renaming `builder` to `Builder` - not required as it's a variable, but might be easier to read to signal it's a variable holding a reference to a class

**Review by halaprix — APPROVED** [review]
left two comments - generally looks good

**Review by halaprix — APPROVED** [review]
left two comments - generally looks good

**Review by halaprix — DISMISSED** [review]

File: sdk/protocol-plugins/src/implementation/BaseProtocolPlugin.ts (hunk 2 above)
Comment (robercano) [reply sdk/protocol-plugins/src/implementation/BaseProtocolPlugin.ts:147]: Agree, I'll change it to BuilderClass

File: sdk/protocol-plugins/src/plugins/aave-v3/builders/AaveV3DepositBorrowActionBuilder.ts (hunk 1 above)
Comment (robercano) [reply sdk/protocol-plugins/src/plugins/aave-v3/builders/AaveV3DepositBorrowActionBuilder.ts:92]: I'll add a switch case for this so I can throw an error on `default`

**Review by halaprix — APPROVED** [review]

**Pull request merged**